File: BeesAlgorithm/src/studies/beesalgortithm/Utilities.java
package studies.beesalgortithm;

import java.util.Collections;
import java.util.Vector;

public class Utilities {
	
	public static Vector<Scout> createScouts(int scoutsAmount){
		Vector<Scout> scouts = new Vector<Scout>(); 
		for(int i=0;i<scoutsAmount;i++){
			scouts.add(new Scout()); 
		}
		return scouts; 
	}
	
	public static void sendScouts(Vector<Scout> scouts) {
		for(Scout scout:scouts){
			scout.choseFlower(); 
		}
	}

	public static void searchEliteFlowersNeigberhood(Vector<Scout> scoutsWithEliteFlowers) {
		Flower bestNeighbor; 
		for(Scout scout : scoutsWithEliteFlowers){
			bestNeighbor = chooseBestNeighbor(scout.getFlower(), BeesAlgorithm.getElisteSolutionsAmount());
			scout = new Scout(bestNeighbor);
		}
	}


	public static void searchBestFlowersNeigberhood(Vector<Scout> scoutsWithBestFlowers) {
		Flower bestNeighbor; 
		for(Scout scout : scoutsWithBestFlowers){
			bestNeighbor = chooseBestNeighbor(scout.getFlower(), BeesAlgorithm.getBestSolutionsAmount());
			scout = new Scout(bestNeighbor);
		}
		
	}
	
	private static Flower chooseBestNeighbor(Flower flower, int neigborsAmount) {
		Vector<Flower> neighborhood = new Vector<Flower>();
		for (int i=0;i<neigborsAmount;i++){
			Flower neighbor = flower.choseFlowerNeighbor();
			neighborhood.add(neighbor); 
		}
		Collections.sort(neighborhood);
		return neighborhood.firstElement();
	}
	
	
}

File: QueueingNetworks/Optimization/Utilities.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Optimization
{
    static class Utilities
    {
        public static List<Scout> createScouts(int scoutsAmount)
        {
            List<Scout> scouts = new List<Scout>();
            for (int i = 0; i < scoutsAmount; i++)
            {
                scouts.Add(new Scout());
            }
            return scouts;
        }

        public static void sendScouts(List<Scout> scouts)
        {
            foreach (Scout scout in scouts)
            {
                scout.choseFlower();
            }
        }

        public static void searchEliteFlowersNeigberhood(List<Scout> scoutsWithEliteFlowers)
        {
            Flower bestNeighbor;
            for(int i = 0; i < scoutsWithEliteFlowers.Count; i++)
            {
                bestNeighbor = chooseBestNeighbor(scoutsWithEliteFlowers[i].getFlower(), BeesAlgorithm.getElisteSolutionsAmount());
                scoutsWithEliteFlowers[i] = new Scout(bestNeighbor);
            }
        }


        public static void searchBestFlowersNeigberhood(List<Scout> scoutsWithBestFlowers)
        {
            Flower bestNeighbor;
            for (int i = 0; i < scoutsWithBestFlowers.Count; i++)
            {
                bestNeighbor = chooseBestNeighbor(scoutsWithBestFlowers[i].getFlower(), BeesAlgorithm.getBestSolutionsAmount());
                scoutsWithBestFlowers[i] = new Scout(bestNeighbor);
            }
        }

        private static Flower chooseBestNeighbor(Flower flower, int neigborsAmount)
        {
            List<Flower> neighborhood = new List<Flower>();
            for (int i = 0; i < neigborsAmount; i++)
            {
                Flower neighbor = flower.choseFlowerNeighbor();
                neighborhood.Add(neighbor);
            }
            neighborhood.Sort();
            return neighborhood.First();
        }
    }
}
